addons/udes_stock/tests/test_picking.py

```diff
@@ -406,3 +406,51 @@ def test09_considers_partially_available_move_lines_when_suggesting_location(sel
 
         self.assertNotIn(picking1.move_line_ids.location_dest_id, suggested_locations)
         self.assertEqual(suggested_locations, self.test_location_05)
+
+class TestPickingWarning(common.BaseUDES):
+    """Test the generation of warning messages for user-set pre-conditions"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestPickingWarning, cls).setUpClass()
+        Picking = cls.env['stock.picking']
+
+        cls.products_info = [{'product': cls.apple, 'qty': 10}]
+        cls.test_picking = cls.create_picking(cls.picking_type_in,
+                                              origin="test_picking_origin",
+                                              products_info=cls.products_info,
+                                              confirm=True)
+        cls.test_picking = cls.test_picking.sudo(cls.inbound_user)
```

Comment:
If we do this at the end of the setUpClass we probably don't need the sudos of lines 426 and 427.

Reply:
Moved